### DISCUSSION TITLE
Raspberry PI 4 losing Output Device setting

### DISCUSSION THREAD

**Discussion by mrManniManfred** [rern/rAudio#225] [Bug Reports]
Hi,
I am using a Raspberry Pi 4 with an HiFiBerry DAC2 HD connected.

After reboot the device is sometimes not active in MPD, instead the on-board headphones are selected.

Sometimes the HiFiBerry DAC2 HD is not listed after reboot but only on-board headphones and HDMI 1.

Using the lastest software update. Is there a good log which I could provide?

Regards

**Comment by rern** [comment]
Remove the DAC from GPIO and reinstall. Make sure it's inserted all the way down.
Try it on another Raspberry Pi if possible.

**Reply by mrManniManfred** [reply]
DAC is sitting tight. Recognized as "hifiberry-dacplushd" ca. 50% of reboots.
Have no other PI right now but don't think it is a hardware issue, when recognized it works fine.
Will try to find some logs :)..

**Reply by AndyFx12** [reply]
I am using a HifiBerry DAC+. Your described issue occurs sometimes on my RPi3, too. In some rare conditions, I had to switch of my power-supply before DAC was found again.
So I decided (cause of daily usage) to keep them powered on. A RPi is known to consume low energy.
regards, Andy

**Comment by R101G** [comment]
I remember having a similar problem too. It might be worth checking that _dtparam=audio=off_ is in config.txt.

**Reply by mrManniManfred** [reply]
With the setting "dtparam=audio=off" there is no output device displayed.

**Reply by R101G** [reply]
So it looks as if the Pi is sometimes not detecting your DAC at all, rather than detecting both DACs and selecting the wrong one, which is the problem I had. You might need to narrow the problem down by checking that the driver is being loaded and that there are no useful messages, using lsmod and dmesg.

If you are not using the latest image, it might be worth trying that, or updating with pacman -Syu, in case something has changed.

**Reply by R101G** [reply]
Another random thought .... 

As it works sometimes, this is likely to be a timing problem. I wonder if the DAC is not powering up as quickly as the Pi? If it never fails on a reboot, but sometimes fails on a power up, that could indicate that the DAC is not always ready when the Pi starts probing.
